src/overlay005/map_prop_animation.c

```diff
@@ -0,0 +1,835 @@
+}
+
+static int MapPropAnimation_IsLooping(const MapPropAnimation *animation)
+{
+    GF_ASSERT(animation->loaded == TRUE);
+    return animation->looping;
+}
+
+MapPropAnimation *MapPropAnimationManager_LoadPropAnimationForOneShot(const int mapPropModelID, const int mapPropAnimationIndex, const int animationLoopCount, const int pastoriaGymButtonGroup, const BOOL animationReversed, const BOOL animationPaused, const BOOL isDeferredLoading, NNSG3dResMdl *mapPropModel, NNSG3dResTex *mapPropTexture, MapPropAnimationManager *manager)
+{
+    MapPropAnimation *animation;
+    int i;
+    int animeArchiveID;
+    MapPropAnimeListFile animeListFile;
+
+    NARC_ReadWholeMember(manager->animeListNARC, mapPropModelID, &animeListFile);
+    GF_ASSERT(mapPropAnimationIndex < MAP_PROP_ANIME_LIST_FILE_ARCHIVE_IDS_COUNT);
+
+    animeArchiveID = animeListFile.animeArchiveIDs[mapPropAnimationIndex];
+
+    if (animeArchiveID == 0xFFFFFFFF) {
+        return NULL;
+    }
+
+    if (MapPropAnimation_CheckDeferredLoadingFlag(animeListFile.flags) != isDeferredLoading) {
+        return NULL;
+    }
+
+    for (i = 0; i < MAP_PROP_ANIMATION_MANAGER_MAX_ANIMATIONS; i++) {
+        if ((pastoriaGymButtonGroup != 0) && (manager->animations[i].pastoriaGymButtonGroup == pastoriaGymButtonGroup)) {
+            GF_ASSERT(FALSE);
+        }
+    }
+
+    for (i = 0; i < MAP_PROP_ANIMATION_MANAGER_MAX_ANIMATIONS; i++) {
+        if (!manager->animations[i].loaded) {
+            manager->animations[i].loaded = TRUE;
+            manager->animations[i].paused = animationPaused;
+            manager->animations[i].animationObj = NULL;
+            manager->animations[i].currentLoop = 0;
+            manager->animations[i].animeArchiveID = animeArchiveID;
+            manager->animations[i].reversed = animationReversed;
+            manager->animations[i].pastoriaGymButtonGroup = pastoriaGymButtonGroup;
+            manager->animations[i].animationFile = MapPropAnimation_LoadAnimationObj(manager->animeNARC, animeArchiveID, mapPropModel, &manager->animations[i].animationObj, mapPropTexture, &manager->allocator);
+            manager->animations[i].loopCount = animationLoopCount;
+            manager->animations[i].looping = TRUE;
+
+            animation = &manager->animations[i];
+
+            MapPropAnimation_GoToFirstFrame(&manager->animations[i]);
+            return animation;
+        }
+    }
+
+    GF_ASSERT(FALSE);
+    return NULL;
+}
+
+void MapPropAnimationManager_LoadPropAnimations(const int mapPropModelID, NNSG3dResMdl *mapPropModel, NNSG3dResTex *mapPropTexture, MapPropAnimationManager *manager)
+{
+    int animeArchiveID;
+    int i;
+    int j;
+    MapPropAnimeListFile animeListFile;
+
+    NARC_ReadWholeMember(manager->animeListNARC, mapPropModelID, &animeListFile);
+
+    // BUG: The generated NARC contains 0xFF instead of 0x00,
+    // so this early exit is never taken
+    if (!animeListFile.hasAnimations) {
+        return;
+    }
```

Comment:
**thought**: Wow! That's an interesting find. Is this worthy of a write-up in `docs/bugs_and_glitches.md`? Or do we want to hold off on it until we can determine what a fix would look like?

Reply:
As a starter: I'm pretty sure this here is a bug because no file in `bm_anime_list.narc` matches either
* `hasAnimations == 0xFF && animeArchiveIDs[0] != 0xFFFFFFFF`, or
* `hasAnimations == 0x01 && animeArchiveIDs[0] == 0xFFFFFFFF`

So, `animeArchiveIDs[0] == 0xFFFFFFFF` always mean `hasAnimations == 0xFF` (which is why I'm pretty sure that `hasAnimations` is the correct name for this field).

So, I think the fix would simply be:
```suggestion
    if (animeListFile.hasAnimations == 0xFF) {
        return;
    }
```

As for writing this in `docs/bugs_and_glitches.md`, I think this bug doesn't really come from the game but rather from the tool GF used to generate that NARC. If we document those as well, then why not ^^

It's also technically a bug you'll never see, as those checks are always (== both times) immediately followed by something like:
```c
if (animeArchiveIDs[i] == 0xFFFFFFFF) {
    return;
}
```

So it may have been overzealous of me to call this a bug, more of an oopsie.